File: 3. Longest Substring Without Repeating Characters/main.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  if (s.length <= 1) return s.length;

  let lp = 0;
  let maxCount = 1;
  let letters = new Set();

  for (let rp = 0; rp < s.length; ++rp) {
    while (letters.has(s[rp])) {
      letters.delete(s[lp]);
      ++lp;
    }

    letters.add(s[rp]);
    maxCount = Math.max(maxCount, rp - lp + 1);
  }

  return maxCount;
};

File: 3. Longest Substring Without Repeating Characters/main.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        letters = set()
        lp, rp, maxLength = 0, 0, 0

        for i in range(len(s)):
            while s[rp] in letters:
                letters.remove(s[lp])
                lp += 1
            
            letters.add(s[rp])
            maxLength = max(maxLength, rp - lp + 1)
            rp +=1

        return maxLength